docs/topics/tasks.txt

```diff
@@ -0,0 +1,335 @@
+
+Enqueueing tasks
+================
+
+To add the task to the queue store, so it will be executed, call the ``enqueue``
+method on it::
+
+    result = calculate_meaning_of_life.enqueue()
+
+Returned is a :class:`django.tasks.TaskResult`, which can be used to retrieve
+the result of the task once it has finished executing.
+
+If the task takes arguments, these can be passed as-is to ``enqueue``::
+
+    result = calculate_meaning_of_life.enqueue(answer=42)
+
+To enqueue tasks in an ``async`` context, :meth:`aenqueue <django.tasks.Task.aenqueue>`
+is available as an ``async`` variant of ``enqueue``.
+
+Transactions
+------------
+
+By default, tasks are enqueued after the current transaction (if there is one)
+commits successfully (using :meth:`transaction.on_commit <django.db.transaction.on_commit>`),
+rather than enqueueing immediately.
+
+This behavior can be changed by changing the :setting:`TASKS-ENQUEUE_ON_COMMIT`
+setting for the backend, or for a specific task using the ``enqueue_on_commit``
+parameter.
+
+Task results
+============
+
+When enqueueing a ``Task``, you receive a :class:`django.tasks.TaskResult`,
+however it's likely useful to retrieve the result from somewhere else (for example
+another request or another task).
+
+Each ``TaskResult`` has a unique :attr:`id <django.tasks.TaskResult.id>`, which
+can be used to identify and retrieve the result once the code which enqueued the
+task has finished.
+
+The :meth:`django.tasks.Task.get_result` method can retrieve a result based on
+its ``id``::
+
+    # Later, somewhere else...
+    result = calculate_meaning_of_life.get_result(result_id)
+
+If the ``TaskResult`` exists, it is returned. If it doesn't exist, or isn't a
+result for ``calculate_meaning_of_life``, :exc:`django.tasks.exceptions.ResultDoesNotExist`
+is raised.
+
+To retrieve a ``TaskResult``, regardless of which kind of ``Task`` it was from,
+use the ``get_result`` method on the API
+
+To retrieve results in an ``async`` context, :meth:`aget_result <django.tasks.Task.aget_result>`
+is available as an ``async`` variant of ``get_result`` on both the backend and ``Task``.
+
+Some backends, such as the built in ``ImmediateBackend`` do not support ``get_result``.
+Calling ``get_result`` on these backends will raise :exc:`NotImplementedError`.
+
+Updating results
+----------------
+
+A ``TaskResult`` contains the status of a task's execution at the point it was
+retrieved. If the task finishes after ``get_result`` is called, it will not update.
+
+To refresh the values, call the :meth:`django.tasks.TaskResult.refresh` method::
+
+    result.status  # RUNNING
+
+    result.refresh()  # or await result.arefresh()
+
+    result.status  # SUCCEEDED
+
+Return values
+-------------
+
+If your task function returns something, it can be retrieved from the
+:attr:`django.tasks.TaskResult.return_value` attribute::
+
+    result.return_value  # 42
+
+If the task has not finished executing, or has failed, :exc:`ValueError` is raised.
```

Comment:
Should the code example above perhaps check for `ResultStatus.SUCCEEDED` before accessing the value, to make it clear the semantics?